flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -412,28 +436,141 @@ public ScheduleMode getScheduleMode() {
 		return scheduleMode;
 	}
 
+	/**
+	 * Scheduling to be performed when an ExecutionVertex is encountered that cannot be resumed
+	 */
+	private interface ScheduleAction {
```

Comment:
Why do theses private interfaces (ScheduleActiona and PostBacktrackingHook) being added here?